### PR TITLE
Disabled buttons and links should not do anything when clicked

### PR DESCRIPTION
**Pull request by forest**

### CODE REVIEW

File: lib/petal_components/link.ex [hunk 1]
```diff
@@ -9,35 +9,55 @@ defmodule PetalComponents.Link do
...
+    assigns = update_in(assigns.rest, &Map.drop(&1, [:"phx-click"]))
+
+    ~H"""
+    <button class={@class} disabled={@disabled} {@rest}>
+      <%= if @label, do: @label, else: render_slot(@inner_block) %>
+    </button>
+    """
+  end
+
+  def a(%{disabled: true} = assigns) do
```
Comment (mplatts) [inline lib/petal_components/link.ex:54]: What if we moved this to the top, and always drop phx-click and always render a button. Then we don't need to change the others as any disabled `a` will hit this first?

Comment (mplatts) [reply lib/petal_components/link.ex:54]: Or, if we still want <a> s to appear, just move both these disabled true function to the top, then don't modify the others

**Comment by forest** [comment]
Good idea. This for sure can be cleaned up. Just want to make sure this is
generally right approach.

File: lib/petal_components/link.ex (hunk 1 above)
Comment (forest) [reply lib/petal_components/link.ex:54]: I think we want both `button` and `a` since those render differently.

**Comment by mplatts** [comment]
Cheers

**Comment by mplatts** [comment]
Released in 1.0.4  https://github.com/petalframework/petal_components/releases/tag/v1.0.4

**Pull request merged**